tools/code-generation/generator/src/main/java/com/amazonaws/util/awsclientgenerator/generators/cpp/CppClientGenerator.java

```diff
@@ -718,15 +718,25 @@ protected SdkFileEntry generateClientSmithyHeaderFile(final ServiceModel service
         VelocityContext context = createContext(serviceModel);
         context.put("CppViewHelper", CppViewHelper.class);
         context.put("RequestlessOperations", requestlessOperations);
-        Optional<String> firstAuthScheme = serviceModel.getAuthSchemes().stream().filter(entry->ResolverMapping.containsKey(entry)).findFirst();
-        if(firstAuthScheme.isPresent())
+        
+        if(serviceModel.getAuthSchemes().size() > 1)
         {
-            context.put("AuthSchemeResolver", ResolverMapping.get(firstAuthScheme.get()));
+            context.put("AuthSchemeResolver", "SigV4MultiAuthSchemeResolver");
+            context.put("IsMultiAuth", true);
         }
         else
         {
-            throw new RuntimeException(String.format("authSchemes '%s'",serviceModel.getAuthSchemes().stream().collect(Collectors.toList())
-            ));
+            Optional<String> firstAuthScheme = serviceModel.getAuthSchemes().stream().filter(entry->ResolverMapping.containsKey(entry)).findFirst();
+
+            if(firstAuthScheme.isPresent())
+            {
+                context.put("AuthSchemeResolver", ResolverMapping.get(firstAuthScheme.get()));
+            }
+            else
+            {
+                throw new RuntimeException(String.format("authSchemes '%s'",serviceModel.getAuthSchemes().stream().collect(Collectors.toList())
+                ));
```

Comment:
fix these trailing params, put them on the same line as everything else, it reads weird